teps/0057-tekton-catalog-pipeline-organization.md

```diff
@@ -0,0 +1,470 @@
+
+### Requirements & Guidelines
+
+Guidelines aim to help users author their tekton
+resource. Requirements are _applying_ those requirements to the
+upstream catalog. The requirements are not part of the Catalog
+contract but they should be applied as much as possible to other
+catalogs.
+
+Automation might be run against all verified and official resources to
+ensure (see [Tekton Catalog Test
+Infrastructure](https://docs.google.com/document/d/1-czjvjfpuIqYKsfkvZ5RxIbtoFNLTEtOxaZB71Aehdg/edit#)):
+
+1. Resources have all description fields filled in
+
+2. Resources must have app.kubernetes.io /version to indicate the
+   version of the resource
+
+3. Resources can include the annotation "tekton.dev/displayName" to
+   indicate the name to display in the Tekton Hub, otherwise the name
+   will be computed based on the task file name.
+
+4. Resources can include annotation "tekton.dev/tags" which is a comma
+   separated tags associated with it e.g. kaniko task can have a tag:
+   container-image, build-tool
+
+5. Resources should include an annotation indicating the minimum
+   version of Tekton Pipelines they are expected to be compatible with
+   (format TBD)
+
+6. Include a README, though they should defer to the resource’s
+   description fields to do most of the heavy lifting. The first line
+   of the description could be the summary, followed by a blank line
+   and then the body akin to a git commit message.
+
+7. They pass [yamllint](https://github.com/tektoncd/catalog/issues/101)
+
+8. Additional yaml requirements can be enforced with [conftest](https://garethr.dev/2019/06/introducing-conftest/) as
+   they are discovered
+
+9. Referenced images are published to public registries and do not
+   contain major known vulnerabilities
+
+10. A [working example](https://docs.google.com/document/d/1pZY7ROLuW47ymZYqUgAbxskmirrmDg2dd8VPATYXrxI/edit?ts=5e559121#heading=h.e9bh8qes6kh6) is included which uses all parameters and configurable values of
+    the resource.
+
+### Deprecation & Removal strategy
+
+There might be some cases where a Resource is no more useful or
+shouldn’t be used any more — and maybe even removed. There needs to be
+some rules that clearly define those cases.
+
+First, we need to define what deprecated and removed
+
+- **Deprecated** means
+
+  - The resource is considered as read-only and should only get
+    security updates (if any)
+
+  - The resource stays available in the catalog and is still tested
+    on compatible versions
+
+- **Removed** means
+
+  - The resource is not available anymore in the catalog
+
+A resource can be deprecated in the following cases
+
+- The tool used by the resource is deprecated itself (e.g. a hub Task
+  would be marked as deprecated when hub is deprecated)
+
+- The resource is taken over by another one with a different name (and
+  different versions, …)
+
+- The resource is not maintained anymore (by the OWNERs) and is not
```

Comment:
Since we don't collect telemetry in Tekton, it might be difficult to know exactly about usage? Unless we have data about downloads from the hub?

If feel like we don't need to call out all specific cases for removal in the TEP, as long as we specify the deprecation nd removal policy  in terms of how much notice we need to give and through which channels.

Reply:
Currently hub doesn't have any data about downloads since we are providing the install command. But it completely make sense to know the usage of each resource